### ISSUE TITLE
Scheduler Task: Default behavior

### ISSUE THREAD

**Issue by zwenn** [cobwebch/external_import#251]
Hej Francois,

first thank you very much for these great importer extensions and the great work of documentation.

I use the newest versions of external_import with svconnector and svconnector_csv in Typo3v10.4.30.

After all configuration I want to import my cdv data via scheduler task and come across the default behavior
of the task and email notification.

In my opinion it should work like this:
- no file found -> do nothing (no error/exception,  no email notification, no logging)

So fist, if no file is found there occurs an error in svconnector/Classes/Utility/FileUtility.php
Function getFileAsTemporaryFile returns only false, so one can't react on this failure.

I add
   // Exit early if file content could not be read
   if ($fileContent === false) {

        /** NEW: set error: File not found */
        $this->setError('File not found');

        return false;
   }
       
In external_import/Classes/Task/AutomatedSyncTask.php
I prevent email notification this way:

        /** NEW: check $messages if there is one with: File not found from svconnector/Classes/Utility/FileUtility.php */
        foreach ($messages as $message) {
            foreach ($message as $value) {
                if (str_contains($value, '(reason: File not found)')) {
                    $fileExists = false;
                }
            }
        }

        // If necessary, prepare a report with all messages
        /** NEW: Report only if file exist */
        if ($fileExists && !empty($extensionConfiguration['reportEmail'])) {
            $reportContent .= $importer->getReportingUtility()->reportForTable(
                $this->table,
                $this->index,
                $messages
            );
            if (count($messages[AbstractMessage::ERROR]) > 0) {
                $globalStatus = 'ERROR';
            } elseif (count($messages[AbstractMessage::WARNING]) > 0) {
                $globalStatus = 'WARNING';
            }
            // Assemble the subject and send the mail
            $subject = empty($extensionConfiguration['reportSubject']) ? '' : $extensionConfiguration['reportSubject'];
            $subject .= ' [' . $globalStatus . '] ' . 'Synchronization of table ' . $this->table . ', index ' . $this->index;
            $importer->getReportingUtility()->sendMail($subject, $reportContent);
        }
    }

    // If any warning or error happened, throw an exception
    /** NEW: no exception if file does not exist in scheduler context */
    if ($fileExists && $globalStatus !== 'OK') {
        throw new \Exception(
            'One or more errors or warnings happened. Please consult the log.',
            1258116760
        );
    }
Maybe there is a nicer solution but works for me.
Maybe you could implement this or something like this, so everyone could use this.

**Comment by fsuter** [comment]
Thanks for the detailed feedback and the suggested solutions. I think the whole reporting should be improved, but I must say it has not been a top priority for now. I can see how it could be useful to have a full report for each import run and choose where it should be sent (to some logging system, by mail, or whatever else).

**Comment by fsuter** [comment]
I got back to this while checking old issues and the first part of your correction does not make sense to me: `getFileAsTemporaryFile` indeed returns early if `$fileContent` it `false` but an error is set inside `getFileContent` which the connector service may retrieve (and does retrieve in the case of "svconnector_csv"). Extract from the code:

```php
if ($temporaryFile === false) {
        $error = $fileUtility->getError();
```

However I found that in the case where an absolute path is used for pointing to the file, no error was reported. Maybe that's what caused your observation. I opened an issue in "svconnector" for this: https://github.com/cobwebch/svconnector/issues/27
